lib/semantics/check-do.cc

```diff
@@ -28,12 +28,34 @@ namespace Fortran::semantics {
 
 using namespace parser::literals;
 
+static NamePtr GetNamePtr(const std::optional<parser::Name> &name) {
```

Comment:
This should be generalized and put into lib/common/template.h among the other std::optional<> utilities if it's useful.

Reply:
Thanks.  For me, it made the code a little more readable.  I'll move it.